### PR TITLE
docs(solana_sdk): Improve docs

### PR DESCRIPTION
**Pull request by guibescos**
Addresses comment about the docs from https://github.com/pyth-network/pyth-crosschain/pull/1408

### CODE REVIEW

File: target_chains/solana/sdk/js/pyth_solana_receiver/src/PythSolanaReceiver.ts [hunk 1]
```diff
@@ -168,15 +181,15 @@ export class PythTransactionBuilder extends TransactionBuilder {
 
   /**
    * Add instructions to update price feed accounts to the builder.
+   * Price feed accounts are a special type of price update accounts. Instead of being ephemeral accounts, they are fixed accounts for each feed id that can only be updated with a more recent price update than the one they contain.
+   * Use this function to update a shared price feed account with a recent price
```
Comment (guibescos) [inline target_chains/solana/sdk/js/pyth_solana_receiver/src/PythSolanaReceiver.ts:185]: Not a 100% satisfied with this explanation

File: target_chains/solana/sdk/js/pyth_solana_receiver/src/PythSolanaReceiver.ts [hunk 2]
```diff
@@ -61,19 +61,26 @@ export type PythTransactionBuilderConfig = {
...
  * A builder class to build transactions that:
- * - Post price updates (fully or partially verified)
+ * - Post price updates (fully or partially verified) or update price feed accounts
  * - Consume price updates in a consumer program
  * - (Optionally) Close price update and encoded vaa accounts to recover the rent (`closeUpdateAccounts` in `PythTransactionBuilderConfig`)
  *
+ * `addPostPriceUpdates` vs `addUpdatePriceFeed`:
+ * - `addPostPriceUpdates` is used to post price updates to ephemeral accounts. Use this to post a price update from the present or from the past for your program to consume.
+ * - `addUpdatePriceFeed` is used to post price updates to price feed accounts, these are fixed accounts for each feed id that can only be updated with a more recent price update than the one they contain. Their addresses can be found using `getPriceFeedAccountAddress`. Use this to update a shared price feed account with a recent price
+ *
```
Comment (jayantk) [inline target_chains/solana/sdk/js/pyth_solana_receiver/src/PythSolanaReceiver.ts:71]: I think you want to move the more abstract discussion to this comment and then have these lower-level comments on the methods.

File: target_chains/solana/sdk/js/pyth_solana_receiver/src/PythSolanaReceiver.ts (hunk 1 above)
Comment (jayantk) [reply target_chains/solana/sdk/js/pyth_solana_receiver/src/PythSolanaReceiver.ts:185]: yeah... I think one way to fix this is to create a richer mental model for the reader to understand how things work. I think the model is something like:

* Pyth allows you to post any price update on-chain. This info is stored in price update accounts
* Pyth additionally supports price feed accounts, which hold a stream of price updates that move forward in time. Programs can read a price feed account to get a recent price for a specific feed id. Price feeds are built on top of pull updates (i.e., using price update accounts).

Key thing to clarify is the relationship between price feeds and price updates. Is a feed composed of price updates? is it a sequence of updatse?

**Review by jayantk — APPROVED** [review]

File: target_chains/solana/sdk/js/pyth_solana_receiver/src/PythSolanaReceiver.ts [hunk 3]
```diff
@@ -704,23 +715,33 @@ export class PythSolanaReceiver {
...
 /**
  * Derive the address of a price feed account
- * @param shardId The shard ID of the set of price feed accounts. This shard ID allows for multiple sets of price feed accounts to be managed by the same program.
+ * @param shardId The shard ID of the set of price feed accounts. This shard ID allows for multiple price feed accounts for the same price feed id to exist.
  * @param priceFeedId The price feed ID.
  * @param pushOracleProgramId The program ID of the Pyth Push Oracle program. If not provided, the default deployment will be used.
  * @returns The address of the price feed account
  */
-function getPriceFeedAccountAddress(
+function getPriceFeedAccountAddressHelper(
```
Comment (guibescos) [inline target_chains/solana/sdk/js/pyth_solana_receiver/src/PythSolanaReceiver.ts:744]: Why did you introduce `getPriceFeedAccountAddressHelper`

File: target_chains/solana/sdk/js/pyth_solana_receiver/README.md [hunk 4]
```diff
@@ -1,116 +1,192 @@
...
+The address of the price feed accounts can be derived automatically from the feed id and the shard id:
 
-```ts
-import { PublicKey } from "@solana/web3.js";
-import { PriceServiceConnection } from "@pythnetwork/price-service-client";
-import { PythSolanaReceiver } from "@pythnetwork/pyth-solana-receiver";
-import { MyFirstPythApp, IDL } from "./idl/my_first_pyth_app";
+```typescript
+const solUsdPriceFeedAccount = pythSolanaReceiver
+  .getPriceFeedAccountAddress(SOL_PRICE_FEED_ID, 1)
```
Comment (guibescos) [inline target_chains/solana/sdk/js/pyth_solana_receiver/README.md:152]: The order seems wrong here

File: target_chains/solana/sdk/js/pyth_solana_receiver/examples/post_price_update.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+const SOL_PRICE_FEED_ID =
+  "0xef0d8b6fda2ceba41da15d4095d1da392a0d2f8ed0c6c7bc0f4cfac8c280b56d";
+const ETH_PRICE_FEED_ID =
+  "0xff61491a931112ddf1bd8147cd1b641375f79f5825126d665480874634fd0ace";
+
+let keypairFile = "";
+if (process.env["SOLANA_KEYPAIR"]) {
+  keypairFile = process.env["SOLANA_KEYPAIR"];
+} else {
+  keypairFile = "~/.config/solana/id.json";
```
Comment (guibescos) [inline target_chains/solana/sdk/js/pyth_solana_receiver/examples/post_price_update.ts:20]: Had to use `os.homedir`, the `~` wasn't working

File: target_chains/solana/sdk/js/pyth_solana_receiver/examples/post_price_update.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+async function main() {
+  const connection = new Connection("https://api.mainnet-beta.solana.com");
```
Comment (guibescos) [inline target_chains/solana/sdk/js/pyth_solana_receiver/examples/post_price_update.ts:24]: This should probably be devnet

File: target_chains/solana/sdk/js/pyth_solana_receiver/examples/post_price_update_instructions.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  );
+
+  // Put your instructions here
+  const consumerInstructions: InstructionWithEphemeralSigners[] = [];
+
+  const transactions = await pythSolanaReceiver.batchIntoVersionedTransactions(
+    [...postInstructions, ...consumerInstructions, ...closeInstructions],
+    { computeUnitPriceMicroLamports: 1000000 }
+  ); // Put all the instructions together
+  await pythSolanaReceiver.provider.sendAll(transactions);
```
Comment (guibescos) [inline target_chains/solana/sdk/js/pyth_solana_receiver/examples/post_price_update_instructions.ts:52]: I had to add  `{preflightCommitment:"processed"}` to get this to land reliably

File: target_chains/solana/sdk/js/pyth_solana_receiver/src/PythSolanaReceiver.ts [hunk 8]
```diff
@@ -168,15 +184,15 @@ export class PythTransactionBuilder extends TransactionBuilder {
 
   /**
    * Add instructions to update price feed accounts to the builder.
+   * Price feed accounts are a special type of price update accounts. Instead of being ephemeral accounts, they are fixed accounts for each feed id that can only be updated with a more recent price update than the one they contain.
+   * Use this function to update a shared price feed account with a recent price
```
Comment (guibescos) [inline target_chains/solana/sdk/js/pyth_solana_receiver/src/PythSolanaReceiver.ts:188]: I still don't like this text that much

**Pull request merged**